### PR TITLE
Showing "Autosave enabled" toast

### LINKED ISSUE #159: Showing a toast "autosave is enabled/disabled"
Make it more clear whether or not autosave is enabled, e.g. display a toast saying that it is, when a project is opened or when a project is saved for the first time.

### PR DESCRIPTION
**Pull request by yining1023**
fixes #159

### CODE REVIEW

File: client/constants.js [hunk 1]
```diff
@@ -98,3 +98,6 @@ export const RESET_PASSWORD_RESET = 'RESET_PASSWORD_RESET';
 export const INVALID_RESET_PASSWORD_TOKEN = 'INVALID_RESET_PASSWORD_TOKEN';
 // eventually, handle errors more specifically and better
 export const ERROR = 'ERROR';
+
+export const JUST_OPEN_PROJECT = 'JUST_OPEN_PROJECT';
```
Comment (catarak) [inline client/constants.js]: For the same of consistency, can this be called "JUST_OPENED_PROJECT"?

File: client/constants.js [hunk 2]
```diff
@@ -98,3 +98,6 @@ export const RESET_PASSWORD_RESET = 'RESET_PASSWORD_RESET';
...
+export const RESET_JUST_OPEN_PROJECT = 'RESET_JUST_OPEN_PROJECT';
```
Comment (catarak) [inline client/constants.js]: and can this be called "RESET_JUST_OPENED_PROJECT"

File: client/modules/IDE/actions/ide.js [hunk 3]
```diff
@@ -195,3 +195,15 @@ export function resetInfiniteLoops() {
     type: ActionTypes.RESET_INFINITE_LOOPS
   };
 }
+
+export function justOpenProject() {
```
Comment (catarak) [inline client/modules/IDE/actions/ide.js]: and this justOpenedProject()

File: client/modules/IDE/actions/ide.js [hunk 4]
```diff
@@ -195,3 +195,15 @@ export function resetInfiniteLoops() {
...
+  return {
+    type: ActionTypes.JUST_OPEN_PROJECT,
+  };
+}
+
+export function resetJustOpenProject() {
```
Comment (catarak) [inline client/modules/IDE/actions/ide.js]: and this resetJustOpenedProject()

File: client/modules/IDE/actions/ide.js [hunk 5]
```diff
@@ -195,3 +195,15 @@ export function resetInfiniteLoops() {
...
+  return {
+    type: ActionTypes.RESET_JUST_OPEN_PROJECT
```
Comment (catarak) [inline client/modules/IDE/actions/ide.js]: this will need to be updated with the new name

File: client/modules/IDE/actions/project.js [hunk 6]
```diff
@@ -5,12 +5,13 @@ import JSZip from 'jszip';
 import JSZipUtils from 'jszip-utils';
 import { saveAs } from 'file-saver';
 import { showToast, setToastText } from './toast';
-import { setUnsavedChanges } from './ide';
+import { setUnsavedChanges, justOpenProject, resetJustOpenProject } from './ide';
```
Comment (catarak) [inline client/modules/IDE/actions/project.js]: this will also need to be updated

File: client/modules/IDE/actions/project.js [hunk 7]
```diff
@@ -5,12 +5,13 @@ import JSZip from 'jszip';
...
 
 const ROOT_URL = location.href.indexOf('localhost') > 0 ? 'http://localhost:8000/api' : '/api';
 
 export function getProject(id) {
   return (dispatch) => {
+    dispatch(justOpenProject());
```
Comment (catarak) [inline client/modules/IDE/actions/project.js]: this as well

File: client/modules/IDE/actions/project.js [hunk 8]
```diff
@@ -52,8 +53,15 @@ export function saveProject(autosave = false) {
             type: ActionTypes.PROJECT_SAVE_SUCCESS
           });
           if (!autosave) {
-            dispatch(showToast());
-            dispatch(setToastText('Project saved.'));
+            if (state.ide.projectJustOpened && state.preferences.autosave) {
```
Comment (catarak) [inline client/modules/IDE/actions/project.js]: can this variable be called state.ide.justOpenedProject

File: client/modules/IDE/actions/project.js [hunk 9]
```diff
@@ -52,8 +53,15 @@ export function saveProject(autosave = false) {
...
+              dispatch(showToast(5500));
+              dispatch(setToastText('Project saved.'));
+              setTimeout(() => dispatch(setToastText('Autosave enabled.')), 1500);
+              dispatch(resetJustOpenProject());
```
Comment (catarak) [inline client/modules/IDE/actions/project.js]: this will need to be updated with the new name

File: client/modules/IDE/actions/project.js [hunk 10]
```diff
@@ -73,8 +81,15 @@ export function saveProject(autosave = false) {
             files: response.data.files
           });
           if (!autosave) {
-            dispatch(showToast());
-            dispatch(setToastText('Project saved.'));
+            if (state.preferences.autosave) {
+              dispatch(showToast(5500));
+              dispatch(setToastText('Project saved.'));
+              setTimeout(() => dispatch(setToastText('Autosave enabled.')), 1500);
+              dispatch(resetJustOpenProject());
```
Comment (catarak) [inline client/modules/IDE/actions/project.js]: this will need to be updated

File: client/modules/IDE/reducers/ide.js [hunk 11]
```diff
@@ -16,7 +16,8 @@ const initialState = {
   unsavedChanges: false,
   infiniteLoop: false,
   previewIsRefreshing: false,
-  infiniteLoopMessage: ''
+  infiniteLoopMessage: '',
+  projectJustOpened: false
```
Comment (catarak) [inline client/modules/IDE/reducers/ide.js]: and this as well

File: client/modules/IDE/reducers/ide.js [hunk 12]
```diff
@@ -79,6 +80,10 @@ const ide = (state = initialState, action) => {
       return Object.assign({}, state, { previewIsRefreshing: true });
     case ActionTypes.END_SKETCH_REFRESH:
       return Object.assign({}, state, { previewIsRefreshing: false });
+    case ActionTypes.JUST_OPEN_PROJECT:
```
Comment (catarak) [inline client/modules/IDE/reducers/ide.js]: and this

File: client/modules/IDE/reducers/ide.js [hunk 13]
```diff
@@ -79,6 +80,10 @@ const ide = (state = initialState, action) => {
...
+      return Object.assign({}, state, { projectJustOpened: true });
+    case ActionTypes.RESET_JUST_OPEN_PROJECT:
```
Comment (catarak) [inline client/modules/IDE/reducers/ide.js]: and, lastly, this

**Comment by catarak** [comment]
Just added a few comments about having consistent naming. I tried to add comments on all the lines you'll need to update so they will be easier not to miss.

**Comment by catarak** [comment]
hmmm I'm getting the error `Actions may not have an undefined "type" property. Have you misspelled a constant?`

**Comment by yining1023** [comment]
So sorry about this. Now it's all good.

**Pull request merged**